src/WebformCivicrmPostProcess.php

```diff
@@ -910,6 +910,11 @@ private function saveGroupsAndTags($contact, $cid, $c) {
         $field_name = 'civicrm_' . $c . '_contact_1_' . $fid;
         if (!empty($contact['other'][1][$type]) || isset($this->enabled[$field_name])) {
           $add = wf_crm_aval($contact, "other:1:$type", array());
+          foreach ($add as $k => $v) {
+            if ($v == 0) {
+              unset($add[$k]);
+            }
+          }
           $remove = empty($this->existing_contacts[$c]) ? array() : $this->getExposedOptions($field_name, $add);
```

Comment:
If you can rewrite this to -> (see below) then I can merge this PR because I already tested this:

```
          // ToDo - $add should only contains the option(s) selected so unset everything else b/c addOrRemoveMultivaluedData is expecting that and we need this to handle TagSets - this is essentially a fail-safe
          foreach ($this->getExposedOptions($field_name) as $k => $v) {
            if ($add[$k] === 0) {
              unset($add[$k]);
            }
          }
```

Reply:
Code changed as requested, and tested OK at my end.